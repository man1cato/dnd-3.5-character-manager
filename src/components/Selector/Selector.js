import React, { useEffect, useState } from 'react'
import _ from 'lodash'

import './Selector.scss'


const Selector = ({ items, Content, handleSelect }) => {	
	const [currentItemId, setCurrentItemId] = useState(items[0].id)

	useEffect(() => {
		setCurrentItemId(items[0].id)
	}, [items])

	return (
		<>
			<div className="Selector__row">
				<select 
					className="Selector__dropdown select"
					value={currentItemId} 
					onChange={(e) => setCurrentItemId(e.target.value)}
				>
					{_.map(items, item => (
						<option 
							value={item.id} 
							key={item.id}
						>
							{item.name}
						</option>
					))}
				</select> 
				
				<button 
					className="Selector__button button"
					type="button" 
					onClick={() => handleSelect(currentItemId)}
				>
					Select
				</button>
			</div>

			<Content
				currentItemId={currentItemId}
			/>
		</>
	)
}

export default Selector