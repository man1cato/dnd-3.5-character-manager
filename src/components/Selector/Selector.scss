@import '../../styles/base/settings';


.Selector__row {
   display: flex;
   justify-content: space-between;
   align-items: center;
   height: 4rem;
   margin-bottom: $m-size;
}

.Selector__dropdown {
   flex: 1 1 75%;
   margin-right: $s-size;
}

.Selector__button {
   flex: 1 0 100px;
}