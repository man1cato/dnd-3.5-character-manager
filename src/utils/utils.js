import _ from 'lodash'


export const convertTextToArray = (string) => _.compact(string.split(/\n/))

export const findItemById = (storeItemArray, itemId) => storeItemArray.find((item) => item.id === itemId)

export const apiObjectToArray = (obj) => _.orderBy(Object.keys(obj).map((id) => ({
	id,
	...obj[id]
})), ['name'], ['asc'])