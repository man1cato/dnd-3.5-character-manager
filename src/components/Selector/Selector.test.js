import React from 'react'
import { render, fireEvent } from '@testing-library/react'

import Selector from './Selector'
import { apiData } from '../../test-utils/utils'
import { apiObjectToArray } from '../../utils/utils'


const handleSelect = jest.fn()
let props
beforeAll(async () => {
   const api = await apiData()
   props = {
      items: apiObjectToArray(api.feats),
      Content: () => (
         <p>
            Description: Feat description
         </p>
      ),
      handleSelect
   }
})

test('should render Selector correctly with feats data', () => {
   const { container } = render(<Selector {...props} />)
   expect(container).toMatchSnapshot()
})