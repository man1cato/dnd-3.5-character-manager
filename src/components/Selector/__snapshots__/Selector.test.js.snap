// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Selector correctly with feats data 1`] = `
<div>
  <div
    class="Selector__row"
  >
    <select
      class="Selector__dropdown select"
    >
      <option
        value="recdmotqOlGgN2otb"
      >
        Ability Focus
      </option>
      <option
        value="recoTlzPQ37bNLFsm"
      >
        Acrobatic
      </option>
      <option
        value="recat3oy3R6kW5Uts"
      >
        Additional Magic Item Space
      </option>
      <option
        value="rec8uIUF1rFAa2VuG"
      >
        Agile
      </option>
      <option
        value="recvzMqdCBIzd7z9n"
      >
        Alertness
      </option>
      <option
        value="recnrQnRgUIJQkJOz"
      >
        Animal Affinity
      </option>
      <option
        value="recST2D4Kyaknrsg3"
      >
        Armor Proficiency (Heavy)
      </option>
      <option
        value="recRXzmlbn1MzfoKr"
      >
        Armor Proficiency (Light)
      </option>
      <option
        value="recIucb9yIBFfwadA"
      >
        Armor Proficiency (Medium)
      </option>
      <option
        value="recTjM3xaRKY5Kzxv"
      >
        Armor Skin
      </option>
      <option
        value="recOCyJT5oWscbchC"
      >
        Athletic
      </option>
      <option
        value="recOm8zpuXUjChQZH"
      >
        Augment Summoning
      </option>
      <option
        value="recBbahLUMGxoCyFF"
      >
        Augmented Alchemy
      </option>
      <option
        value="recEjwHNjDZrJG5mP"
      >
        Automatic Quicken Spell
      </option>
      <option
        value="recVKDhL94EQG8Bhm"
      >
        Automatic Silent Spell
      </option>
      <option
        value="recMYg4ohuZoVhHg3"
      >
        Automatic Still Spell
      </option>
      <option
        value="recQyW6ByR4ClPt80"
      >
        Autonomous
      </option>
      <option
        value="recwouXSFSvKDipwo"
      >
        Awesome Blow
      </option>
      <option
        value="reckdrHwbucszWlx3"
      >
        Bane of Enemies
      </option>
      <option
        value="recjk6hkPLOgMt6eb"
      >
        Blind-Fight
      </option>
      <option
        value="recX0vHtJoi5E4lcm"
      >
        Blinding Speed
      </option>
      <option
        value="recDGl9s66lwqKgCb"
      >
        Blindsight, 5-ft. Radius
      </option>
      <option
        value="recZkUp3LzpK7Yl5e"
      >
        Bonus Domain
      </option>
      <option
        value="recnxtKAeEeCyobZv"
      >
        Brew Potion
      </option>
      <option
        value="recLqr9Z8dkls4ZO1"
      >
        Bulwark of Defense
      </option>
      <option
        value="recoGrm1vruan9LwF"
      >
        Chaotic Rage
      </option>
      <option
        value="recBl7E7mkej8Kh4a"
      >
        Cleave
      </option>
      <option
        value="rec6j8vDL2r3twF8o"
      >
        Cloak Dance
      </option>
      <option
        value="reclpbMRlP2bTvA8K"
      >
        Colossal Wild Shape
      </option>
      <option
        value="recKz8Kl8cTQvocaG"
      >
        Combat Archery
      </option>
      <option
        value="recdR990pCoRoCtHN"
      >
        Combat Casting
      </option>
      <option
        value="recrc1dWL5N3RPKd6"
      >
        Combat Expertise
      </option>
      <option
        value="recXnkMszr2a4cMG2"
      >
        Combat Reflexes
      </option>
      <option
        value="recVBmMvMyOEvRtJs"
      >
        Craft Cognizance Crystal
      </option>
      <option
        value="rec2DWXl3MtnjXqRQ"
      >
        Craft Construct
      </option>
      <option
        value="recSUUSgdqMnZhJnH"
      >
        Craft Dorje
      </option>
      <option
        value="recaofYxttizBhOSa"
      >
        Craft Epic Magic Arms and Armor
      </option>
      <option
        value="recxra1jpE6L3EIU4"
      >
        Craft Epic Rod
      </option>
      <option
        value="recB1ADsi6OGKayaa"
      >
        Craft Epic Staff
      </option>
      <option
        value="recXxMDo6R93rlTxl"
      >
        Craft Epic Wondrous Item
      </option>
      <option
        value="recXP8I7doU9uFI0W"
      >
        Craft Magic Arms and Armor
      </option>
      <option
        value="recUprO478F0lrfWL"
      >
        Craft Psicrown
      </option>
      <option
        value="recfSnDJwFVsQAdyB"
      >
        Craft Psionic Arms and Armor
      </option>
      <option
        value="recOEPO2WXGIKsXqc"
      >
        Craft Psionic Construct
      </option>
      <option
        value="recXKTZ8KtdrXujgn"
      >
        Craft Rod
      </option>
      <option
        value="recxQ92YDwvBv8JJy"
      >
        Craft Staff
      </option>
      <option
        value="recyMC6cyMOMLHIws"
      >
        Craft Universal Item
      </option>
      <option
        value="rec8F85455BwZqMwF"
      >
        Craft Wand
      </option>
      <option
        value="recRjuAWe7oy6a7wY"
      >
        Craft Wondrous Item
      </option>
      <option
        value="rechTdfI8DKVXUOuJ"
      >
        Damage Reduction
      </option>
      <option
        value="recI7IQe71f44n0qs"
      >
        Deadly Precision
      </option>
      <option
        value="recUuhHOIgnt9V7tt"
      >
        Deafening Song
      </option>
      <option
        value="recg1OIceV2bKe80O"
      >
        Death of Enemies
      </option>
      <option
        value="recFrheLpwNhWFzS3"
      >
        Deceitful
      </option>
      <option
        value="reciyf1VerwnZ0ARP"
      >
        Deflect Arrows
      </option>
      <option
        value="recPlnmL0xLwCkhqW"
      >
        Deft Hands
      </option>
      <option
        value="recPBNJBz2lwwuRyy"
      >
        Devastating Critical
      </option>
      <option
        value="recIZdOr3DGPdpbc8"
      >
        Dexterous Fortitude
      </option>
      <option
        value="recfjtRjVXdn3xLDZ"
      >
        Dexterous Will
      </option>
      <option
        value="rec2vFKZkwMh178E7"
      >
        Diehard
      </option>
      <option
        value="rec83JXxeqFaq13F1"
      >
        Diligent
      </option>
      <option
        value="rechBgTFWqu5YnOc1"
      >
        Diminutive Wild Shape
      </option>
      <option
        value="recP76MPxeKkTmFDW"
      >
        Dire Charge
      </option>
      <option
        value="recsvpX6XhsOvLE4O"
      >
        Disguise Spell
      </option>
      <option
        value="rec7EdrAu7sixASam"
      >
        Distant Shot
      </option>
      <option
        value="reccKhXP1P03au3WT"
      >
        Divine Might
      </option>
      <option
        value="recrA225bMNGuLGF2"
      >
        Divine Vengeance
      </option>
      <option
        value="recyqJvG85XCUZG4x"
      >
        Dodge
      </option>
      <option
        value="rec35lBICs5ipkw7m"
      >
        Dragon Wild Shape
      </option>
      <option
        value="recCT1vB6wSaLMWxi"
      >
        Efficient Item Creation
      </option>
      <option
        value="recKYm3IPVLJ115Bb"
      >
        Empower Spell
      </option>
      <option
        value="recHnrZHpmp7psvj2"
      >
        Empower Spell-Like Ability
      </option>
      <option
        value="recVCgn24xVSXigbM"
      >
        Endurance
      </option>
      <option
        value="reclmfhLdJV7LOrS2"
      >
        Energy Resistance
      </option>
      <option
        value="recYUF3ddygflrQ8w"
      >
        Energy Substitution
      </option>
      <option
        value="recnrsa2FZ2yaKVXn"
      >
        Enhance Spell
      </option>
      <option
        value="recmf4osOlKrZtp6W"
      >
        Enlarge Spell
      </option>
      <option
        value="recTHyYmqyReN98vw"
      >
        Epic Dodge
      </option>
      <option
        value="recbfQSkxH5dfDswW"
      >
        Epic Endurance
      </option>
      <option
        value="reckenqb745nxu31H"
      >
        Epic Fortitude
      </option>
      <option
        value="recAT8p4Gs8rj67Rf"
      >
        Epic Inspiration
      </option>
      <option
        value="recBSUl9GE1MUb6YH"
      >
        Epic Leadership
      </option>
      <option
        value="recPEtaL0j3Ooo8Mj"
      >
        Epic Manifestation
      </option>
      <option
        value="reclI3nmYiSQAFp9T"
      >
        Epic Prowess
      </option>
      <option
        value="rec2mAkEGs7m5RoNr"
      >
        Epic Reflexes
      </option>
      <option
        value="rec1xvVMTsxgPGHpY"
      >
        Epic Reputation
      </option>
      <option
        value="reccuF0PtktDswV4T"
      >
        Epic Skill Focus
      </option>
      <option
        value="recaqg6GFLvmQKzJW"
      >
        Epic Speed
      </option>
      <option
        value="reciXQdgT1ysY46vz"
      >
        Epic Spell Focus
      </option>
      <option
        value="recKVzySxSc8PjyuG"
      >
        Epic Spell Penetration
      </option>
      <option
        value="rec36m2Rxdw3ffRjC"
      >
        Epic Spellcasting
      </option>
      <option
        value="rec74fL73E22QlwJB"
      >
        Epic Toughness
      </option>
      <option
        value="recYH63fal3NkDHgm"
      >
        Epic Weapon Focus
      </option>
      <option
        value="recwJazo0cOaHVN82"
      >
        Epic Weapon Specialization
      </option>
      <option
        value="recSDxh2UD49CVRyh"
      >
        Epic Will
      </option>
      <option
        value="rec1UNqoxirOSxnno"
      >
        Eschew Materials
      </option>
      <option
        value="recmjqppdD8oDAHp6"
      >
        Exceptional Deflection
      </option>
      <option
        value="recfeBOXTX2up9aUs"
      >
        Exotic Weapon Proficiency
      </option>
      <option
        value="recnVpaSB379EQSI0"
      >
        Extend Spell
      </option>
      <option
        value="recNPOssjXXPtbKbb"
      >
        Extended Life Span
      </option>
      <option
        value="recGByPLOPCGQZEW1"
      >
        Extra Music
      </option>
      <option
        value="recMrrf3flxHUSgMR"
      >
        Extra Turning
      </option>
      <option
        value="recuXAYNPNco1uMdG"
      >
        Eyes in the Back of Your Head
      </option>
      <option
        value="recHRtjo6zEHSw7aL"
      >
        Familiar Spell
      </option>
      <option
        value="recVXoEU44zDtbech"
      >
        Far Shot
      </option>
      <option
        value="recHkaFWSLu7SI1ee"
      >
        Fast Healing
      </option>
      <option
        value="recjTsJRbGi8Lptpa"
      >
        Fine Wild Shape
      </option>
      <option
        value="rec8icuDiUWvX1Yry"
      >
        Fleet of Foot
      </option>
      <option
        value="recw2WZMbQ3gonDew"
      >
        Flyby Attack
      </option>
      <option
        value="reczzAAR51o1WRmdB"
      >
        Forge Epic Ring
      </option>
      <option
        value="recohPSgPy9lyD80w"
      >
        Forge Ring
      </option>
      <option
        value="rectdd0HKAYm9SmIx"
      >
        Gargantuan Wild Shape
      </option>
      <option
        value="recCcqyJ5GL9nrx7G"
      >
        Great Charisma
      </option>
      <option
        value="reclT78Am0M14ogH9"
      >
        Great Cleave
      </option>
      <option
        value="recG6zBgLZVpCVJ5Y"
      >
        Great Constitution
      </option>
      <option
        value="recVCGoQYyuB3ywo9"
      >
        Great Dexterity
      </option>
      <option
        value="recboOAfWHQTIZGfr"
      >
        Great Fortitude
      </option>
      <option
        value="recIo1dWJeopo2HWe"
      >
        Great Intelligence
      </option>
      <option
        value="recX34PyAdbEQghsb"
      >
        Great Smiting
      </option>
      <option
        value="recDgjKlw3qhC6Caz"
      >
        Great Strength
      </option>
      <option
        value="reckTAO6zjLWF7qZm"
      >
        Great Wisdom
      </option>
      <option
        value="recngsilwkNCpwJ1w"
      >
        Greater Manyshot
      </option>
      <option
        value="recLKX0TNjlgzGUOx"
      >
        Greater Multiweapon Fighting
      </option>
      <option
        value="recfy4jWSGtP77EQv"
      >
        Greater Spell Focus
      </option>
      <option
        value="recyUNLnXQEp7Qams"
      >
        Greater Spell Penetration
      </option>
      <option
        value="recyONl4n3KUvcZMu"
      >
        Greater Two-Weapon Fighting
      </option>
      <option
        value="recvX8MNxV9RUcIHx"
      >
        Greater Weapon Focus
      </option>
      <option
        value="recgDuc3CRemo6gop"
      >
        Greater Weapon Specialization
      </option>
      <option
        value="rec0o0ZQvthxaPWsK"
      >
        Group Inspiration
      </option>
      <option
        value="recDanI1EtcjYcvaT"
      >
        Heighten Spell
      </option>
      <option
        value="recFR8hcOZ1VWJeSB"
      >
        Hindering Song
      </option>
      <option
        value="recKZ213PbqbXUlZE"
      >
        Hold the Line
      </option>
      <option
        value="reczLPssqes9Allcm"
      >
        Holy Strike
      </option>
      <option
        value="recEepS7K6IXvQOQU"
      >
        Hover
      </option>
      <option
        value="recRflwPRNsV8Mikq"
      >
        Ignore Material Components
      </option>
      <option
        value="recB5m8zpzGVdfA47"
      >
        Imprint Stone
      </option>
      <option
        value="recH0yaYZUkBoX9Ts"
      >
        Improved Alignment-Based Casting
      </option>
      <option
        value="recH507n9eFXFuMtT"
      >
        Improved Arrow of Death
      </option>
      <option
        value="recfGJQ6pkNUqa9MX"
      >
        Improved Aura of Courage
      </option>
      <option
        value="recgOxsvTHb6WuRpx"
      >
        Improved Aura of Despair
      </option>
      <option
        value="recdPR3fvAC3RKGZV"
      >
        Improved Bull Rush
      </option>
      <option
        value="recQwtCq32BVgmIpa"
      >
        Improved Combat Casting
      </option>
      <option
        value="recIGMapWpvesoRZd"
      >
        Improved Combat Reflexes
      </option>
      <option
        value="recEpNvwwJ71UvJm5"
      >
        Improved Counterspell
      </option>
      <option
        value="rec5slnNRWWJC8fb2"
      >
        Improved Critical
      </option>
      <option
        value="recOw2MnHi1Umie9f"
      >
        Improved Darkvision
      </option>
      <option
        value="recQc3GpnKelT57Gs"
      >
        Improved Death Attack
      </option>
      <option
        value="recFWf1n5dhhmR308"
      >
        Improved Disarm
      </option>
      <option
        value="recPwR8YqatQuJqXS"
      >
        Improved Elemental Wild Shape
      </option>
      <option
        value="recQDaucsIOGOt9hp"
      >
        Improved Familiar
      </option>
      <option
        value="recm090Df8faJMFly"
      >
        Improved Favored Enemy
      </option>
      <option
        value="recJrvSJJCUuc6mhl"
      >
        Improved Feint
      </option>
      <option
        value="recZtJj4tIw5XDZbC"
      >
        Improved Flyby Attack
      </option>
      <option
        value="reczT5FbnHSASKG1t"
      >
        Improved Grapple
      </option>
      <option
        value="recmAqqlY2XXd6vQp"
      >
        Improved Heighten Spell
      </option>
      <option
        value="reciazjHVWQmiq1jP"
      >
        Improved Initiative
      </option>
      <option
        value="recReZmh5AR841RDE"
      >
        Improved Ki Strike
      </option>
      <option
        value="recIgfqdlPFEQaYpG"
      >
        Improved Low-Light Vision
      </option>
      <option
        value="recbK0MkITKWaBVvR"
      >
        Improved Manyshot
      </option>
      <option
        value="recRctlW1vIRg7aJE"
      >
        Improved Metamagic
      </option>
      <option
        value="rec7w1n3zWvQgybB5"
      >
        Improved Multiattack
      </option>
      <option
        value="recpqEOvW44yc6IG5"
      >
        Improved Multiweapon Fighting
      </option>
      <option
        value="recJSbPjUKOTWNe6m"
      >
        Improved Natural Armor
      </option>
      <option
        value="recNxmdwGsIHTAPxj"
      >
        Improved Natural Attack
      </option>
      <option
        value="recrcXqW0aKN7CRNg"
      >
        Improved Overrun
      </option>
      <option
        value="recMPvjjsCrx8n3MH"
      >
        Improved Precise Shot
      </option>
      <option
        value="recEqV253RXFXXTeO"
      >
        Improved Shield Bash
      </option>
      <option
        value="recLvwm9VX1l0QOAF"
      >
        Improved Sneak Attack
      </option>
      <option
        value="rec8SML7edkgm8Daa"
      >
        Improved Spell Capacity
      </option>
      <option
        value="recZ7I3tsw8Wglevc"
      >
        Improved Spell Resistance
      </option>
      <option
        value="recBIq6dCa2hCQdXi"
      >
        Improved Stunning Fist
      </option>
      <option
        value="rec4xrHW8QTf6eTe3"
      >
        Improved Sunder
      </option>
      <option
        value="recv8vlK0if2wVkM2"
      >
        Improved Trip
      </option>
      <option
        value="recuEsiEZ1O2huYCr"
      >
        Improved Turning
      </option>
      <option
        value="recspc34ofcNegBbe"
      >
        Improved Two-Weapon Fighting
      </option>
      <option
        value="recCmnimojxd0MNLS"
      >
        Improved Unarmed Strike
      </option>
      <option
        value="recFLUTUFiAZrtWc8"
      >
        Improved Whirlwind Attack
      </option>
      <option
        value="rec1XRG78wqr7t77E"
      >
        Incite Rage
      </option>
      <option
        value="recopIF1mvoCADeqG"
      >
        Infinite Deflection
      </option>
      <option
        value="recSsoZCkJeVaJa2j"
      >
        Inspire Excellence
      </option>
      <option
        value="recEQuzMOnWrYWbWM"
      >
        Instant Reload
      </option>
      <option
        value="recCqxdMJjuhG8R7d"
      >
        Intensify Spell
      </option>
      <option
        value="rec3uAWs5IilEadt9"
      >
        Intuitive Trapfinding
      </option>
      <option
        value="recs5pgQMU6zotYqL"
      >
        Investigator
      </option>
      <option
        value="rec0Rmpc0o4zr5KQZ"
      >
        Iron Will
      </option>
      <option
        value="recBS1Vlwp32LWNNe"
      >
        Jack of All Trades
      </option>
      <option
        value="recAeGlhBNBh7upAs"
      >
        Keen Strike
      </option>
      <option
        value="recCxUg2DPCVGd5wQ"
      >
        Knock-Down
      </option>
      <option
        value="recF1jiy7Z3atlHDa"
      >
        Lasting Inspiration
      </option>
      <option
        value="recaVxDe7IV4Ibppd"
      >
        Leadership
      </option>
      <option
        value="rec37Jez2aiDnkjqE"
      >
        Legendary Climber
      </option>
      <option
        value="rec5gS9hXRPNT4kuC"
      >
        Legendary Commander
      </option>
      <option
        value="recbVb12m3xhnyaHT"
      >
        Legendary Leaper
      </option>
      <option
        value="recd9irRuDR82SnIL"
      >
        Legendary Rider
      </option>
      <option
        value="recbTWNXoiYl9oMVA"
      >
        Legendary Tracker
      </option>
      <option
        value="rec0IcSNwYsg80Ldu"
      >
        Legendary Wrestler
      </option>
      <option
        value="recHPtIzQdLipkdLi"
      >
        Lightning Reflexes
      </option>
      <option
        value="rec3TmlD78AdiPphf"
      >
        Lingering Damage
      </option>
      <option
        value="rec43FVgR2bo1YDR6"
      >
        Magical Aptitude
      </option>
      <option
        value="recExtvt4xN6i7AiO"
      >
        Magical Beast Companion
      </option>
      <option
        value="recq2EZvZ93PkGmh4"
      >
        Magical Beast Wild Shape
      </option>
      <option
        value="rec0Ga2RlLe6NZbhu"
      >
        Manyshot
      </option>
      <option
        value="recu6KSoLrL5T47wp"
      >
        Martial Weapon Proficiency
      </option>
      <option
        value="recXrpfxPke8r5kvy"
      >
        Master Staff
      </option>
      <option
        value="recd4WdR6P3ld60lO"
      >
        Master Wand
      </option>
      <option
        value="recKfsZpmul8Mc89v"
      >
        Maximize Spell
      </option>
      <option
        value="reclfpVhNiab9k5gb"
      >
        Mighty Rage
      </option>
      <option
        value="recUv7HQzFXwLgRB9"
      >
        Mind Over Body
      </option>
      <option
        value="recx6OyqdQn5OUCyB"
      >
        Mobile Defense
      </option>
      <option
        value="rec1oOI1ICj2HuKqa"
      >
        Mobility
      </option>
      <option
        value="recqxhXBJB1KCxzIn"
      >
        Mounted Archery
      </option>
      <option
        value="reciJuIqTWUQRhYv0"
      >
        Mounted Combat
      </option>
      <option
        value="recKkMD7JxJW1XkHj"
      >
        Multiattack
      </option>
      <option
        value="rec43FIS2LihZiFYJ"
      >
        Multispell
      </option>
      <option
        value="recvJkbFXbH5ZXAXa"
      >
        Multiweapon Fighting
      </option>
      <option
        value="recUlskP03KLuUhgq"
      >
        Multiweapon Rend
      </option>
      <option
        value="reclFLacz9XUqkOue"
      >
        Music of the Gods
      </option>
      <option
        value="reccG488HS3PmSuGK"
      >
        Natural Spell
      </option>
      <option
        value="rec5982GoHsn1tfis"
      >
        Negative Energy Burst
      </option>
      <option
        value="recoQZX87YxSgmV2T"
      >
        Negotiator
      </option>
      <option
        value="recL3E07iXqlgd6Mv"
      >
        Nimble Fingers
      </option>
      <option
        value="recFp03E0YFKaq5xr"
      >
        Open Minded
      </option>
      <option
        value="rec7KdtT38XqMZ97s"
      >
        Overwhelming Critical
      </option>
      <option
        value="rec7k1FJnYts9LVOo"
      >
        Penetrate Damage Reduction
      </option>
      <option
        value="rec73w62ZOJtanhuE"
      >
        Perfect Health
      </option>
      <option
        value="rec66QzlM01qvGwRN"
      >
        Perfect Multiweapon Fighting
      </option>
      <option
        value="recCVQuiZAIWOEuJ1"
      >
        Perfect Two-Weapon Fighting
      </option>
      <option
        value="recprHHtBY1m3W8CK"
      >
        Permanent Emanation
      </option>
      <option
        value="recaTDqd3gilGLAc0"
      >
        Persistent Spell
      </option>
      <option
        value="recPjLbGCj3xRNPU3"
      >
        Persuasive
      </option>
      <option
        value="rec7FtveUXIuh63pp"
      >
        Planar Turning
      </option>
      <option
        value="recQV754iVrjVSj66"
      >
        Plant Control
      </option>
      <option
        value="reco0bwyops62ZxPn"
      >
        Plant Defiance
      </option>
      <option
        value="rec81hWf4GNmLBenB"
      >
        Plant Wild Shape
      </option>
      <option
        value="rec0Aa93zpU5V3Asy"
      >
        Point Blank Shot
      </option>
      <option
        value="recxPPaGukcQBMN4X"
      >
        Polyglot
      </option>
      <option
        value="recg6CSotqvsTrakw"
      >
        Positive Energy Aura
      </option>
      <option
        value="recuv5RdScK7Vqhf5"
      >
        Power Attack
      </option>
      <option
        value="recvGUzVy6V0nxd2G"
      >
        Power Critical
      </option>
      <option
        value="recXyoCOASsqgZXOJ"
      >
        Precise Shot
      </option>
      <option
        value="recpHwtjV89aVIoGB"
      >
        Psionic Affinity
      </option>
      <option
        value="recLp3uvkYGZGyywW"
      >
        Quick Draw
      </option>
      <option
        value="recGQ0aIb0LViNIDf"
      >
        Quicken Spell
      </option>
      <option
        value="rec6QsVR3a7aEc8Ev"
      >
        Quicken Spell-Like Ability
      </option>
      <option
        value="rec3Mczx6e5dOlHOM"
      >
        Ranged Inspiration
      </option>
      <option
        value="recjCiBAjZ2MCw8v6"
      >
        Rapid Inspiration
      </option>
      <option
        value="recmNzyXLWdoJ94Vp"
      >
        Rapid Metabolism
      </option>
      <option
        value="recEk5kUnIREk59lF"
      >
        Rapid Reload
      </option>
      <option
        value="recoZbyD7qsch7xSe"
      >
        Rapid Shot
      </option>
      <option
        value="reckz7xRmxxdTbh4G"
      >
        Reach Spell
      </option>
      <option
        value="recii9zxr6QoJFGgb"
      >
        Reactive Countersong
      </option>
      <option
        value="rec3Z5EqrUQXIcAjy"
      >
        Reckless Offense
      </option>
      <option
        value="recVoMid9syjVPQAm"
      >
        Reflect Arrows
      </option>
      <option
        value="recBVTl3j7Rfs1TjG"
      >
        Repeat Spell
      </option>
      <option
        value="reckCFbXtuDrf5Q4N"
      >
        Ride-By Attack
      </option>
      <option
        value="recrcCBspMITAt03y"
      >
        Righteous Strike
      </option>
      <option
        value="recIpizicw40PTLoT"
      >
        Ruinous Rage
      </option>
      <option
        value="rec02pDGILJ5EZz1W"
      >
        Run
      </option>
      <option
        value="recpoSavlX3cOnlk4"
      >
        Sacred Spell
      </option>
      <option
        value="recPdYl618loaB28p"
      >
        Scribe Epic Scroll
      </option>
      <option
        value="recGkxwkqppy7EYmI"
      >
        Scribe Scroll
      </option>
      <option
        value="recBQeNMELDj3IE7A"
      >
        Scribe Tattoo
      </option>
      <option
        value="recrXs4R086L9BQ2v"
      >
        Self-Concealment
      </option>
      <option
        value="recBV4PEpBYb8mVC0"
      >
        Self-Sufficient
      </option>
      <option
        value="rechWZl17jELhDThh"
      >
        Sharp-Shooting
      </option>
      <option
        value="recy4KGNJkJPrSK6f"
      >
        Shattering Strike
      </option>
      <option
        value="recm19w48xltockBW"
      >
        Shield Proficiency
      </option>
      <option
        value="recQNpK7311nbBt15"
      >
        Shot on the Run
      </option>
      <option
        value="recghbS5KnXshwJmR"
      >
        Sidestep Charge
      </option>
      <option
        value="recVyjh0BfUwtXfIF"
      >
        Silent Spell
      </option>
      <option
        value="rec2wriPyA1jtpZih"
      >
        Simple Weapon Proficiency
      </option>
      <option
        value="recaj8lyaZepMlFCM"
      >
        Skill Focus
      </option>
      <option
        value="recpeswIYdGxGtd4g"
      >
        Snatch
      </option>
      <option
        value="recU0tKxbZB3ShosK"
      >
        Snatch Arrows
      </option>
      <option
        value="recNul9zU9EzV9JYC"
      >
        Sneak Attack of Opportunity
      </option>
      <option
        value="recaMu31K8bE507Qh"
      >
        Spectral Strike
      </option>
      <option
        value="recr648zasZ6BHY5D"
      >
        Spell Focus
      </option>
      <option
        value="recJTQZBWe3uN9DKV"
      >
        Spell Knowledge
      </option>
      <option
        value="rec4KkfhKpetKIDqj"
      >
        Spell Mastery
      </option>
      <option
        value="recTphofr2OGPwf06"
      >
        Spell Opportunity
      </option>
      <option
        value="recQM5J2Z4Bg345O0"
      >
        Spell Penetration
      </option>
      <option
        value="recm8xrzST82XRtbb"
      >
        Spell Stowaway
      </option>
      <option
        value="recJJTjUAlnPTK5lI"
      >
        Spellcasting Harrier
      </option>
      <option
        value="recY4eMlPUaTvuZVL"
      >
        Spirited Charge
      </option>
      <option
        value="rec0xSHVkvnocAPyx"
      >
        Spontaneous Domain Access
      </option>
      <option
        value="reczBTYLm1NMiiWKO"
      >
        Spontaneous Spell
      </option>
      <option
        value="recj9JGm6nU9sSM3W"
      >
        Spring Attack
      </option>
      <option
        value="recB7JU0g2Bfz5xqS"
      >
        Stand Still
      </option>
      <option
        value="rece4EQ3n0Ya33LGf"
      >
        Stealthy
      </option>
      <option
        value="recCEnHpjCHJli3pq"
      >
        Still Spell
      </option>
      <option
        value="recE6PzbZgtMuUcVm"
      >
        Storm of Throws
      </option>
      <option
        value="recJeI9Rlq6R1io9r"
      >
        Stunning Fist
      </option>
      <option
        value="recZ74DWSuSZULLZh"
      >
        Subdual Substitution
      </option>
      <option
        value="recfSU9k2yDFrOJl1"
      >
        Superior Expertise
      </option>
      <option
        value="recDB8YX4av29yRxm"
      >
        Superior Initiative
      </option>
      <option
        value="recfzLBBH3LdkdqGM"
      >
        Swarm of Arrows
      </option>
      <option
        value="recGWkniPVHTwxxDB"
      >
        Tenacious Magic
      </option>
      <option
        value="receeaaQusxqJ2ZF7"
      >
        Terrifying Rage
      </option>
      <option
        value="recBowYlyXstRa97P"
      >
        Thundering Rage
      </option>
      <option
        value="reczbG3R5q7p4lj9R"
      >
        Toughness
      </option>
      <option
        value="rec4VSjFtUxZwK886"
      >
        Tower Shield Proficiency
      </option>
      <option
        value="recxkSuS8IvkzRVyV"
      >
        Track
      </option>
      <option
        value="recaut9Caobtz40s0"
      >
        Trample
      </option>
      <option
        value="reclhvvxKZVEyX745"
      >
        Two-Weapon Defense
      </option>
      <option
        value="recePeBndah0BTNs4"
      >
        Two-Weapon Fighting
      </option>
      <option
        value="rec9H5Qje89lulBtp"
      >
        Two-Weapon Rend
      </option>
      <option
        value="recwC0QFCIsNCYwCZ"
      >
        Uncanny Accuracy
      </option>
      <option
        value="recJFvWvKKUkM6NJM"
      >
        Undead Mastery
      </option>
      <option
        value="recpwqw7H735wrK81"
      >
        Unholy Strike
      </option>
      <option
        value="reccZ5xjJ9eWUxvY8"
      >
        Vermin Wild Shape
      </option>
      <option
        value="recBnl6h3mATsgZJc"
      >
        Vorpal Strike
      </option>
      <option
        value="recVdnKmMFz54fNLF"
      >
        Weapon Finesse
      </option>
      <option
        value="recy5YNMs3lo9rYTp"
      >
        Weapon Focus
      </option>
      <option
        value="recCUKNPq8gkgp8Pt"
      >
        Weapon Specialization
      </option>
      <option
        value="recCT26q7TKufh4x9"
      >
        Whirlwind Attack
      </option>
      <option
        value="recX8EJKHZx7Qv1yO"
      >
        Widen Aura of Courage
      </option>
      <option
        value="recF264keoM3c2uYX"
      >
        Widen Aura of Despair
      </option>
      <option
        value="recxFliEDF2AXl0rI"
      >
        Widen Spell
      </option>
      <option
        value="recF2gHjeFMKLdUmE"
      >
        Wild Talent
      </option>
      <option
        value="recI40vCPdTE5LsEu"
      >
        Wingover
      </option>
      <option
        value="recgApBVCXci88ubj"
      >
        Zone of Animation
      </option>
    </select>
    <button
      class="Selector__button button"
      type="button"
    >
      Select
    </button>
  </div>
  <p>
    Description: Feat description
  </p>
</div>
`;
